### ISSUE TITLE
Entering a hashtag in the explore search field and pressing enter won't show content with that hashtag

### ISSUE THREAD

**Issue by ell1e** [mastodon/mastodon#22463]
### Steps to reproduce the problem

1. Click "Explore" button
2. Enter a hashtag into the search bar and press enter:

![Screenshot from 2022-12-18 22-36-51](https://user-images.githubusercontent.com/64124388/208320893-67e97b2c-3d5b-417f-8922-3a1835685a59.png)

### Expected behaviour

The "All" tab should show posts first. The "Posts" tab should show actual posts with this hashtag, yet shows "Could not find anything for these search terms".

### Actual behaviour

"All" shows OTHER hashtags first, I doubt this is what the majority expects searching for a full hashtag. "Posts" doesn't even work, just gives an error. In overall, searching for a hashtag just feels plain broken.

### Specifications

Mastodon as running on mastodon.social

**Comment by supernovae** [comment]
I agree, if an instance has elasatic search enabled, the #hashtag search should search all the users toots, favorite, boosts or bookmarks that include that #hashtag

**Comment by ell1e** [comment]
I don't know what exactly elastic search provides, but since `/tags/<tag>` shows posts for that tag I assume the mechanism to provide the results I'm suggesting is available *somehow*. Otherwise I don't know how that other page would work.

**Comment by Atari-Frosch** [comment]
ACK for todon.eu. Same happens even if one adds a space after the searched hashtag. It seems that this space gets removed before starting the actual search.

**Comment by nemobis** [comment]
> "All" shows other hashtags first which I doubt this is what the majority expects.

This is also filed as https://github.com/mastodon/mastodon/issues/22346 and https://github.com/mastodon/mastodon/issues/19575

**Comment by ell1e** [comment]
Those tickets seem to be about what hashtags are returned, not that maybe someone entering a hashtag doesn't want to primarily see other hashtags to start with.
